hmt_escrow/storage.py

```diff
@@ -58,24 +48,53 @@ def download(key: str, private_key: bytes) -> Dict:
     Raises:
         Exception: if reading from IPFS fails.
 
+    >>> credentials = {
+    ... 	"gas_payer": "0x1413862C2B7054CDbfdc181B83962CB0FC11fD92",
+    ... 	"gas_payer_priv": "28e516f1e2f99e96a48a23cea1f94ee5f073403a1c68e818263f0eb898f1c8e5"
+    ... }
+    >>> pub_key = b"2dbc2c2c86052702e7c219339514b2e8bd4687ba1236c478ad41b43330b08488c12c8c1797aa181f3a4596a1bd8a0c18344ea44d6655f61fa73e56e743f79e0d"
+    >>> job = Job(credentials=credentials, escrow_manifest=manifest)
+    >>> (hash_, manifest_url) = upload(job.serialized_manifest, pub_key)
+    >>> manifest_dict = download(manifest_url, job.gas_payer_priv)
+    >>> manifest_dict == job.serialized_manifest
+    True
     """
+
+    # Q prefix is ipfs. else assume ipns
+    if key[0] != 'Q':
```

Comment:
Where did this `Q` come from?

Reply:
we want to see if it is an ipns id. https://github.com/ipfs/faq/issues/22
all ipfs hashes are prefixed with Q. i updated the code for clarity